### ISSUE TITLE
CircleCI Bug

### ISSUE THREAD

**Issue by raen79** [oblador/loki#290]
Hi everyone,

I have been running into an interesting bug on CircleCI with loki for react native. It seems some tests diffs end up in other tests in CircleCI but not locally. Also it seems like the diff is much more sensitive on CircleCI than locally which leads to even more issues. I attempted to make the tolerance 100 on looks-same but this is still not enough, and there are still diffs from uncorrelated tests appearing in each other. Does anyone have any with this issue?

Below is a little decorator I created that has helped a lot, but obviously not solved the issue:

```ts
import React, { FC, useEffect } from 'react';
import type { Props } from './types';

const AsyncStory: FC<Props> = ({ children, onComplete }) => {
  useEffect(() => {
    setTimeout(onComplete, 500);
  }, [onComplete]);

  return <>{children}</>;
};

export default AsyncStory;
```

**Comment by techeverri** [comment]
Loki includes a package that helps with async stories: `@loki/create-async-callback`
🔗 https://loki.js.org/flaky-tests.html#asynchronous-stories

**Comment by raen79** [comment]
@techeverri Yes it is used along with the `AsyncStory` component. However the issue remains. Do you use CircleCI with loki react native?

**Comment by techeverri** [comment]
I've used Loki using Jenkins and Travis.

**Comment by raen79** [comment]
@techeverri So the pinned code was an attempt at solving the issue. The issue I am running into is that Loki seems to be diffing screenshots from different tests in CI. I attempted to amend this by increasing the time between tests (even up to 10 seconds), but nothing seems to do it.

**Comment by techeverri** [comment]
What do you mean by "Loki seems to be diffing screenshots from different tests in CI"?

Do you have a link to an example build that you could share?

**Comment by raen79** [comment]
It's not specifically about any build. But for example let's say that I have story A, story B, and story C. Locally, loki will go through story A, then B, and then C, taking a snapshot of each and storing it. So far so good.

In CI, it will go through A, and compare it to the reference image A. Then it will go to B, and compare it to reference A. Is this clearer?

**Comment by techeverri** [comment]
Yes, now I understand. Thank you for elaborating 👍
I have used Loki for a couple of years and never heard or found this issue 🤔

Let's hope someone else has seen this before, with luck they've found a fix as well.
